### ISSUE TITLE
Competitive analysis (unmoderated usability testing) of CKAN websites - Userflows 

### ISSUE THREAD

**Issue by Aparna1Gopal** [hackforla/access-the-data#191]
### Overview

- Conducting a Competitive Usability Evaluation of existing CKAN Websites 
- Testing the user flows of specific user personas (data expert and data novice)
- Related to Issue: Website User Flows and Wireflows #145
- Related to Issue: Gap Analysis for current User Flows #192

### Action Items

PHASE 1
- [x] Identify the CKAN competitor websites to conduct the tests (3-5 websites) (Showcase websites: https://ckan.org/showcase)
- [x] Identify viable free tool with maximum features to conduct the automated quantitative usability testing (updated: MS Clarity and/or Usability Hub, also Google Analytics with MS Clarity)
- [x] Embed MS Clarity for heat mapping of CKAN websites, for automated quantitative usability testing
- [x] Embed Google Analytics - data on MS Clarity - on CKAN websites, for automated quantitative usability testing
- [x] Data Collection (over 6-8 weeks for preliminary analytics using MS Clarity
- [x] Create and complete Guide/Manual, for unmoderated usability testing. This goal of this guide is to create a replicable process for conducting unmoderated usability testing across ATD. 

PHASE 2 
[Phase 2 was Not implemented since, (i) organic visitors to the websites did not yield statistically significant results, (ii) MS Clarity using Google Tag Manager was integrated on third-party CKAN websites. As a result, full access was not available on the back-end, which also skewed the results from data collection]
- [ ] ~~Analyze collected data for best practices~~ 
- [ ] I~~dentify and collate research findings~~  
- [ ] ~~Draft Slide Deck Template to present research findings~~  
- [ ] ~~Identify and collate Research Findings: Identify the Benchmarking criteria, Best Practices for AtD's website redesign (based on CKAN websites' performance)~~ 
- [ ] ~~Present results & findings to the ATD Team~~ 

DOCUMENTATION
- [x] Create and complete Wiki Issue and Wiki Page
- [x] Create unmoderated usability testing Guide/Manual using MS Clarity, for future testing projects
- [x] Link to [Wiki](https://github.com/hackforla/access-the-data/wiki/ATD-Research:-Unmoderated-Usability-Testing-(Competitive-analysis-of-CKAN-websites))
- [x] [Guide to unmoderated usability testing](https://docs.google.com/document/d/1xHk--1ogLAgqv_zS38caiJkyXZtd6uNdqryJPnxpffc/edit)

**Notes:** 
- [x] Concurrent steps/ Next steps: Conduct Preference Testing: SUS/SUPR-Q surveys of existing CKAN websites to identify usability gaps (based on personas, can use UsabilityHub) (Quantitative) (Conducted by UX Designer on ATD's Design Team)
- [x] Next steps: Conduct Card Sorts to identify usability gaps (in Information Architecture of User Flows) (based on personas) (Qualitative) [ON HOLD] (Can be conducted once the persona research is complete)
- [x] When AtD's testable live website #187 is ready, repeat above action items (to test against benchmarked items) [ON HOLD] [ALTERNATIVELY], (Usability HuB has an option where Figma prototype can be incorporated for preliminary usability testing, can explore this if it is available in the free version - In talks with PMs and UX Design Lead on incorporating this)

### Resources/Instructions
https://github.com/Aparna1Gopal/UXR-TRAINING-NOTES/issues/11
https://github.com/Aparna1Gopal/UXR-TRAINING-NOTES/issues/14
https://github.com/Aparna1Gopal/UXR-TRAINING-NOTES/issues/12
https://github.com/Aparna1Gopal/UXR-TRAINING-NOTES/issues/17
Benchmarking: https://www.nngroup.com/articles/benchmarking-ux/
Google Drive: https://drive.google.com/drive/folders/104KQ5XHPpWXpDcGhwoMJFPwXOqlmUxOQ
Wiki: To create and update
Wiki: To create and update
Wiki Title: Unmoderated usability testing, execution of automated tools
Resources for Wiki MS Clarity 
Clarity Documentation https://docs.microsoft.com/en-us/clarity/
Google Analytics data on MS Clarity https://docs.microsoft.com/en-us/clarity/ga4-integration
Google Tag Manager (GTM) and GDPR compliance https://support.google.com/tagmanager/answer/9323295?hl=en

TEMPLATE FOR PROJECT PROGRESS

Progress: "What is the current status of your project? What have you completed and what is left to do?"
Blockers: "Difficulties or errors encountered."
Availability: "How much time will you have this week to work on this issue?"
ETA: "When do you expect this issue to be completed?"
Pictures: "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
**Notes:** 

Competitor websites identified based on PMs' product vision. Contacted PM (Lucy C & AJ) via Slack. Below are the websites to be tested on: 

https://africaopendata.org/
https://hri.fi/en_gb/
https://data.gov/
https://data.humdata.org/

I am going to be adding https://data.gov.sg, as well due to their active use of iconography/favicons on their website.

**Comment by Aparna1Gopal** [comment]
**Notes:** 

Access the Data Website: 
http://accessthedata.org

Access the Data Test website (Figma embed in Userflow website tool):

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"
- Created Hotjar account and did some navigating around/training to understand how it works. 
- Audited how the external CKAN websites could be embedded into Hotjar to assess usability and user flows for the two personas 
- Audited how Hotjar could be embedded in the Access the Data website to assess usability and user flows in the future as well
- Communicated with Archana M (Design team) if there were overlaps. Since I am only working on the user flows for two specific personas and two specific task questions, we determined there was no overlap. 

Tasks to be completed in the next two weeks: 

- [ ] Complete Research Plan for this particular issue 
- [ ] Complete mini-survey to embed in Hotjar before the CKAN is audited/heat-mapped. Need to see if Hotjar allows a Survey before this heat-mapping (I would like for sections to be together as it would make it for users testing and confuse them less) 
- [ ] Send this draft to PMs (Lucy C, AJ), UX Leads (Mariana V, Kevin C) for feedback 
- [ ] Also pilot this draft within the ATD team via Slack
- [ ] Make revisions based on pilot and feedback 
- [ ] Send the Hotjar CKAN survey out to the world wide web (Recruitment for users via external slack channels) 

Blockers: "Difficulties or errors encountered."
I have not had much time to work on this issue for past few days as I was focussing more on the persona related issues (data expert and community activist). However, the next two weeks will be laser focussed on completing the above mentioned tasks of this issue. 

Availability: "How much time will you have this week to work on this issue?"
6-8 hours 

ETA: "When do you expect this issue to be completed?"
Nest two weeks for Phase 1 

Pictures: "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
**Notes**
**Communication on Slack**

Communication with PM Lucy C, Aparna G (UX Researcher) and Archana (UX Designer)

_Main points_ 
PM Lucy C: We need to create separate GitHub issues so they can track the differences and progress
Aparna G, UXR: Working on this issue #191
Archana, UXD : Working on auditing the UI and design elements. Adding the SUS audit to this work.


_Communication from Slack_
[Aparna Gopal]
Hi aj,
Hope you well! I see you have assigned Archana to issue #191. Could you please clarify what exactly you have in mind as to the direction we could take?
As Archana and I have communicated in the above messages to see if there were any overlaps, but there seemed to be none. And more importantly, I am only testing for two very specific questions related to userflows. This may not cover the full breadth of what the design team currently needs (and what Archana might be looking at). Thing is, I hesitate to add more questions to our users in this particular heatmap survey since we are already testing on 4/5 CKAN websites and anything more than two questions will confuse the user. And they might drop-off before completing the tasks/test. What are your thoughts? I am happy to meet this week to discuss further, if required. Thanks! 

[Aparna Gopal]
One idea I could think of was that Archana could do a complete site audit of the selected 4/5 CKAN websites using a SUS scale. It would need to be audited against a scale to see how these websites are currently performing. This would give us quick and early results. Instead of heatmapping (#issue 191) for all relevant questions that the design team might have and which might take a while since we can deploy only 2-3 survey questions at a time.
In the meantime, I could test with the users (#issue 191) and this would provide richer insights when overlapped with the above SUS audit. I am just brainstorming here! What are all your thoughts and feedback?


[Lucy Chang]
thanks for the details and steps forwards 
[Aparna Gopal]. We put Archana on your issue cause we knew that she was working on something similar but couldnt find an associated issue. 
[Archana]
 what were your plans for the CKAN website audit? I do like Aparna's idea of SUS scale audit. If that fits your plans, it would be great if you could create your own issue for it. otherwise we do have other work that 
[Kevin Chu] and I can direct you towards

[Archana]
Hi 
[Lucy Chang]
 and 
[Aparna Gopal]
, I will be able to work on the SUS audit.
Since Aparna was working on UX research, I decided to audit the UI and design elements. I will add the SUS audit to my work. Thank you!

[Lucy Chang]
that sounds great 
[Archana]
! please make a github issue so we can track the differences and progress


[Aparna Gopal]
[Lucy Chang]
[Archana]
 Thanks for the clarification and feedback! All noted!

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

Created Hotjar account and embedded CKAN websites to test the workings/navigation of HotJar. Also hit a blocker with the free version of Hotjar. 

Able to incorporate only the overall website heat maps with HotJar (will give the overall CKAN analytics) 

Exploring UsabilityHub to incorporate simple survey with user flow testing (2 survey questions) with external users. Even the resources to this might be limited due to free version. 

Communicated with PMs (Lucy C and aj) and Archana M (Design team) if there were overlaps. Immediate questions were clarified for progress. 

Immediate Tasks to be completed:

- [ ]   Complete Research Plan for this particular issue
- [ ]   Complete embedding CKAN websites in HotJar for overall analytics and heatmaps
- [ ]   Start automated data collection on Hotjar 
- [ ]   Complete mini-survey to embed in UsabilityHub before the CKAN is tested for flows. 
- [ ]   Send this draft to PMs (Lucy C, AJ), UX Leads (Mariana V, Kevin C) for feedback
- [ ]   Also pilot this draft within the ATD team via Slack
- [ ]   Make revisions based on pilot and feedback
- [ ]   Send the UsabilityHub CKAN survey out to the world wide web (Recruitment for users via external slack channels)
- [ ]   Wait to receive responses from research participants. Wait 1-2 weeks (10-15 research participants) before commencing data analysis.

Blockers: "Difficulties or errors encountered."
Due to free versions of HotJar and UsabilityHub, the access to all services on each tool is limited. Navigating around each by budgeting the resources available on each tool

Availability: "How much time will you have this week to work on this issue?"
6-8 hours 

ETA: "When do you expect this issue to be completed?"
Next two weeks for Phase 1

Pictures: "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Immediate Tasks to be completed:

- [ ]   Complete Research Plan for this particular issue [Overall Phase 1, 2, 3]
- [ ]   Complete embedding CKAN websites in HotJar for overall analytics and heatmaps
- [ ]   Start automated data collection on Hotjar 
- [ ]   Complete mini-survey to embed in UsabilityHub before the CKAN is tested for flows. 
- [ ]   Send this draft to PMs (Lucy C, AJ), UX Leads (Mariana V, Kevin C) for feedback
- [ ]   Pilot this draft within the ATD team via Slack
- [ ]   Make revisions based on pilot and feedback
- [ ]   Send the UsabilityHub CKAN survey out to the world wide web (Recruitment for users via external slack channels)
- [ ]   Wait to receive responses from research participants. Wait 1-2 weeks (10-15 research participants) before commencing data analysis.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"
Embedded MS  Clarity on the 5 CKAN Websites. Data tracking commenced. Will be waiting and collecting data for the next two weeks. 

Blockers: "Difficulties or errors encountered."
Learning MS Clarity as a tool and learning to understand the needs of the tool for effective quantitative data collection 

Availability: "How much time will you have this week to work on this issue?"
6-8 hours 

ETA: "When do you expect this issue to be completed?"
2-4 weeks based on the data collection 

Pictures: "Add any pictures of the visual changes made to the site so far."
Will be making a slide deck (1 page) to show the progress of the embedding of the CKANs into MS Clarity

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

- Completed the Steps of embedding MS Clarity on the CKAN Websites. Data Collection in progress. 
- Have also gone one step further and embedded Google Analytics that is available with MS Clarity to further collect analytics on the CKAN websites 
- While the above will suffice for automated usability testing, it is also heavily dependant on the number of users visiting these CKAN websites. 
- Hence to enrich the data collection, usability hub is also being used a secondary tool for first click tests to determine the user flow. This will also include recruiting external research participants and the method of data collection varies from MS Clarity and Google Analytics, even though this also functions as another type of unmoderated, automated usability testing. 
- Also cleaned up the Action Items on this issue to keep it simple and also reflect the current progress and path of this issue
- Will be adding access the data email ID as one of the team members on the MS Clarity data collections. Such that PMs and UX Leads can also view the data collection process. 
- Note: The following CKAN websites are being studied for CKAN incorporation and use. 
https://africaopendata.org/
https://hri.fi/en_gb/
https://data.gov/
https://data.humdata.org/
https://data.gov.sg/,

Blockers: "Difficulties or errors encountered."
Just a steep learning curve. But used Google search and available guides (MS Clarity and Google Analytics) to navigate through

Availability: "How much time will you have this week to work on this issue?"
As mentioned to the PMs and UX Leads I will be traveling for the next two weeks. However, data collection is set-up and will be ready to analyze after my return. 

ETA: "When do you expect this issue to be completed?"
Within the next 3-5 weeks (Including research setup, data collection and data analysis) 

Pictures: "Add any pictures of the visual changes made to the site so far."
Adding pictures of the MS Clarity embedded on CKAN Website 





If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"
The automated data collection is currently under progress. 
Added AccesstheData email ID as one of members of the MS Clarity Team 
Have added PM's email ID as one of the member of the MS Clarity Team as requested 

Blockers: "Difficulties or errors encountered."
None 

Availability: "How much time will you have this week to work on this issue?"
Not applicable 

ETA: "When do you expect this issue to be completed?"
We shall wait until the last week of September to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have visited the CKAN websites. 

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable 

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

The automated data collection on selected CKAN websites is currently under progress.

Blockers: "Difficulties or errors encountered."
None

Availability: "How much time will you have this week to work on this issue?"
Not applicable

ETA: "When do you expect this issue to be completed?"
We shall wait until the last week of September/first week of October (another 3-4 weeks) to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have visited the CKAN websites.

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

The automated data collection on selected CKAN websites is currently under progress.

Blockers: "Difficulties or errors encountered."
None

Availability: "How much time will you have this week to work on this issue?"
Not applicable

ETA: "When do you expect this issue to be completed?"
We shall wait until the first week/second week of October (another 2-3 weeks) to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have visited the CKAN websites.

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

The automated data collection on selected CKAN websites is currently under progress.
Discussed the progress on this issue (methods, preliminary insights) during ATD Demo Day in September'22

Blockers: "Difficulties or errors encountered."
None

Availability: "How much time will you have this week to work on this issue?"
Not applicable

ETA: "When do you expect this issue to be completed?"
We shall wait until the second week of October (another 2-3 weeks) to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have organically visited the CKAN websites. 

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

The automated data collection on selected CKAN websites is currently under progress.
Discussed the progress on this issue (methods, preliminary insights) during ATD Demo Day in September'22

Blockers: "Difficulties or errors encountered."
None

Availability: "How much time will you have this week to work on this issue?"
Not applicable

ETA: "When do you expect this issue to be completed?"
We shall wait until the third week of October (post Oct 17) to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have organically visited the CKAN websites. 

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

The automated data collection on selected CKAN websites is currently under progress.
Discussed the progress on this issue (methods, preliminary insights) during ATD Demo Day in September'22

Blockers: "Difficulties or errors encountered."
None

Availability: "How much time will you have this week to work on this issue?"
Not applicable

ETA: "When do you expect this issue to be completed?"
We shall wait until the end of October (Oct Week 4) to review and analyze the first set of data. This way we should have some good quality data with more number of users, who have organically visited the CKAN websites.

Pictures: "Add any pictures of the visual changes made to the site so far."
Not applicable

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
**Notes:** 

- On Demo Day (September 28,2022), UX Researcher discussed the process of using MS Clarity to collect the data on these selected CKAN websites (advantages, disadvantages, reflections etc.)
- Met with Judson L from Dev to discuss the process of embedding tracking code using Google Tag Manager (Dev had follow-up questions from Demo Day) 
- MS Clarity allows tracking analytics data on third part websites (in this case CKAN websites). Guide given here: 
- However, Dev requires the tracking code to be embedded on the backend to track this data. This is only possible if one has ownership of the website and one is able to manually add this 'head code' 
- Discussion between Dev and UXR was to see if Google Tag Manager (& installing MS Clarity) on third-party websites was indeed doing the job of tracking the data correctly. 
- This discussion brought up the need to have a more continuous and regular channel of collaboration between UXR, Dev, and Design teams and not to work in silos. Any UXR project mounted must also keep in mind as to how the end results/findings will better support and aid the Design and Dev teams.

**Comment by Aparna1Gopal** [comment]
**Notes:** 

Reflections an takeaways from working on this issue: 

- Improved process of conducting unmoderated usability tests and automating process. Developed guides, which can be replicated while testing on ATD's website when it is up and running. 
 
- Better and enhanced collaboration between internal stakeholders before mounting any UX Research project. Identity what type of data/insight will be helpful to PMS, Devs, Designers.- actionable insights that can be applied to ATD (example: completing tasks faster, easier navigation etc.) 
 
- Before mounting/undertaking any UXR project, also account for any upcoming challenges (example: limited time, limited budget, faster turnaround time etc.) before selecting research methods for implementation.

**Comment by Aparna1Gopal** [comment]
**Notes: 
One on one with Lucy C, PM on November 14, 2022 (Monday)**

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"
https://github.com/hackforla/access-the-data/issues/191#issuecomment-1317135232

Blockers: "Difficulties or errors encountered."
None 
Currently re-working and re-strategizing based on feedback during one-on-one meeting with PM, based on ATD product vision 

Availability: "How much time will you have this week to work on this issue?"
3-6 hours 

ETA: "When do you expect this issue to be completed?"
2 weeks 

Pictures: "Add any pictures of the visual changes made to the site so far."

If you need help, be sure to either:

ask for help at your next meeting
put a "Status: Help Wanted" label on your issue and pull request
put up a request for assistance on the #access-the-data channel.

**Comment by Aparna1Gopal** [comment]
**Notes**

- [ ] Create Guide for the process of conducting unmoderated usability testing for Access the Data 
- [ ] Update Wiki 
- [ ] at message the PMs regarding the completion 
- [ ] Close issue

**Comment by Aparna1Gopal** [comment]
Progress: "What is the current status of your project? What have you completed and what is left to do?"

- [ ]  Create Guide for the process of conducting unmoderated usability testing for Access the Data
- [ ]  Update Wiki
- [ ]  at message the PMs regarding the completion
- [ ]  Close issue
- [ ] Review the Research Overview link on Wiki, Is the link incorrect? "This page is a subset of our [Research Overview](https://github.com/hackforla/access-the-data/wiki/ATD-Team-Research-Plans) page"

Blockers: "Difficulties or errors encountered."
None
Availability: "How much time will you have this week to work on this issue?"
2-3 hours 
ETA: "When do you expect this issue to be completed?"
End of this week 
Pictures: "Add any pictures of the visual changes made to the site so far."

**Comment by mxajPrice** [comment]
This is complete- thank you!
